### PR TITLE
Some fixed as description

### PR DESCRIPTION
**Pull request by dimaslanjaka**
<details>
  <summary>PR1</summary>

  PR: https://github.com/hexojs/hexo-front-matter/pull/77
  ## chore(regex): fixed parsing for CRLF
  fix for sometimes changing post from github directly
  which doesnt have auto trim whitespaces while saving file
</details>

<details>
  <summary>PR2</summary>

  PR: https://github.com/hexojs/hexo-front-matter/pull/76

  ## fix: types node
  install `@types/node` instead override `type: node`. this method will fixed typescript build for both _single project_ or _workspaces/monorepo project_.
  - https://github.com/dimaslanjaka/hexo-front-matter/commit/68b5eef00b2c593378a7d7f4d8f0695b71726b2b
  - https://github.com/dimaslanjaka/hexo-front-matter/commit/2f1c469a83c44c9cfd91d994d7c5644b428004b2

  ## refactor: enable `esModuleInterop`
  -   the ES6 modules spec states that a namespace import (`import * as x`) can only be an object, by having TypeScript treating it the same as `= require("x")` then TypeScript allowed for the import to be treated as a function and be callable. That's not valid according to the spec.

  -   while accurate to the ES6 modules spec, most libraries with CommonJS/AMD/UMD modules didn't conform as strictly as TypeScript's implementation.

  Turning on `esModuleInterop` will fix both of these problems in the code transpiled by TypeScript. [see more](https://www.typescriptlang.org/tsconfig#esModuleInterop)
  - https://github.com/hexojs/hexo-front-matter/pull/70/commits/2f1c469a83c44c9cfd91d994d7c5644b428004b2#diff-b55cdbef4907b7045f32cc5360d48d262cca5f94062e353089f189f4460039e0

  ## fix: failed build caused by missing lib
  Skip type checking of declaration files from `node_modules`. This can save time during compilation at the expense of type-system accuracy.
  - https://github.com/dimaslanjaka/hexo-front-matter/commit/03fc40d272367f83ce9b0234288735c40a98739c

  ## chore: separate result type and make `options` as optional
  separate result type
  - https://github.com/hexojs/hexo-front-matter/commit/493fd35100f350d7239f8080ff698693863498a5
  - https://github.com/hexojs/hexo-front-matter/commit/594844a55b7261244398648939a6e19102e64210
  optional `options` by adding default `options` an `empty object`
  - https://github.com/dimaslanjaka/hexo-front-matter/blob/f15a9220ac1107bf287cb41c3fae51d895528d19/lib/front_matter.ts#L45

  ## docs: add JSDoc
  - add JSDoc to HexoJS post metadata type
</details>

## refactor: implement `lint-staged` and `husky`
to automatically fix code style before commited (temporary _disabled_)
- https://github.com/dimaslanjaka/hexo-front-matter/commit/6ec68d7d737dd12f3441b7093af745e0b416677c
- https://github.com/dimaslanjaka/hexo-front-matter/commit/87dd66e6ce413194f04b7c4c1bd7b9a204a7ddbf
- https://github.com/dimaslanjaka/hexo-front-matter/commit/5c2e8c5c3fa1712c0ddddc71c9b75182939e34bf

### new changes
- drop `prettier` from `lint-staged` config
- disable automatic init `husky` and `lint-staged` (we can manually activate or deactivate husky)

### CODE REVIEW

**Comment by dimaslanjaka** [comment]
@SukkaW review pls

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
```
Comment (yoshinorin) [inline .husky/pre-commit:2]: This pre-commit hook does not work on Windows. All repositories are tested on cross-platform including Windows.

File: lint-staged.config.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/**
+ * @type {{ [key: string]: import("lint-staged").Commands | import("lint-staged").ConfigFn }}
+ */
+const config = {
+  '*.ts': 'eslint --fix',
+  '*.js': 'eslint --fix',
+  '*.json': 'npx prettier --write',
+  '*.yml': 'npx prettier --write'
```
Comment (yoshinorin) [inline lint-staged.config.js:8]: We have considered using `prettier`, but it is still in draft. If we decide to use it, we will need to update `eslint-config-hexo` for uniformalization.

 - https://github.com/hexojs/eslint-config-hexo/pull/25
 - https://github.com/hexojs/hexo/pull/3719

**Review by yoshinorin — CHANGES_REQUESTED** [review]
Would you please separate the pull request into two parts? type fixes and the others (such as prettier and husky).

To be honest, there is a possibility that we cannot accept the installation of `prettier` and `husky`.

File: .husky/pre-commit (hunk 1 above)
Comment (dimaslanjaka) [reply .husky/pre-commit:2]: but works in my windows.

image

Comment (dimaslanjaka) [reply .husky/pre-commit:2]: successful commit
image

Comment (dimaslanjaka) [reply .husky/pre-commit:2]: btw iam currently using `node` v14 and `lint-staged & husky` latest version.

File: lint-staged.config.js (hunk 2 above)
Comment (dimaslanjaka) [reply lint-staged.config.js:8]: > Would you please separate the pull request into two parts? type fixes and the others (such as prettier and husky).
> 
> To be honest, there is a possibility that we cannot accept the installation of `prettier` and `husky`.

okay ill separate the PR

**Comment by dimaslanjaka** [comment]
i closed this before separate PR

**Comment by dimaslanjaka** [comment]
> Would you please separate the pull request into two parts? type fixes and the others (such as prettier and husky).
> 
> To be honest, there is a possibility that we cannot accept the installation of `prettier` and `husky`.

first PR to fix failed build https://github.com/hexojs/hexo-front-matter/pull/76

File: .husky/pre-commit (hunk 1 above)
Comment (dimaslanjaka) [reply .husky/pre-commit:2]: - disabled auto init husky https://github.com/dimaslanjaka/hexo-front-matter/commit/a5fbaa3e6f7221e6a2e3a5558a0692e58ee17e73

we can manually enable or disable auto linter.

- drop `prettier`

remove prettier from `lint-staged` which still unsupported

File: lint-staged.config.js (hunk 2 above)
Comment (dimaslanjaka) [reply lint-staged.config.js]: already separated. please check https://github.com/hexojs/hexo-front-matter/pull/71#issuecomment-1546599676

File: .husky/pre-commit (hunk 1 above)
Comment (dimaslanjaka) [reply .husky/pre-commit:2]: check i've re-edit my PR descriptions

**Comment by dimaslanjaka** [comment]
already separated PR, pls review again

**Pull request closed without merging**